ocaml/lambda/translcore.ml

```diff
@@ -985,6 +985,45 @@ and transl_exp0 ~in_new_scope ~scopes sort e =
     let l = transl_exp ~scopes sort e in
     if Config.stack_allocation then Lexclave l
     else l
+  | Texp_src_pos ->
+    let fields =
+      Datarepr.labels_of_type
+        Predef.path_lexing_position
+        Predef.lexing_position_decl
+      |> Array.of_list
+      |> Array.map (fun (_, lbl) ->
+          let pos = e.exp_loc.loc_start in
+          let value, typ =
+            let mk_int value =
+              Texp_constant (Const_int value), Predef.type_int
+            in
+            match lbl.lbl_name with
```

Comment:
Doing these string comparisons is unnecessary: we always know the sequence of labels and in what order they will appear.